Update main references in histogram tutorial (#598)
mozilla/data-docs: src/datasets/main_ping_tables.md

# Main Ping Tables

As described in the [pipeline schemas deployment docs](https://docs.telemetry.mozilla.org/concepts/pipeline/schemas.html),
data engineering has a process of generating schemas for pings and deploying them to BigQuery. The Main Ping table (`telemetry.main`)
is one of those generated tables.

> As the number of telemetry probes defined in Firefox grows, so does the number of columns in `telemetry.main`. At this point, we have nearly 10,000 columns, and we ingest many terabytes of main ping data per day. This combination of a very wide schema and a high data volume means that [BigQuery has to reference metadata for a very large number of files](https://console.cloud.google.com/support/cases/detail/25679061?project=moz-fx-data-shared-prod) each time it runs a query, even if it only ends up needing to read a small fraction of those files. This has led to a problematic experience for iterative analysis use cases.

Reviewer: ```suggestion
As the number of telemetry probes defined in Firefox grows, so does the number of columns in `telemetry.main`. As of January 2021, we have nearly 10,000 columns, and we ingest many terabytes of main ping data per day. This combination of a very wide schema and a high data volume means that [BigQuery has to reference metadata for a very large number of files](https://console.cloud.google.com/support/cases/detail/25679061?project=moz-fx-data-shared-prod) each time it runs a query, even if it only ends up needing to read a small fraction of those files. This has led to a problematic experience for iterative analysis use cases.
```

# Main Ping Tables

As described in the [pipeline schemas deployment docs](https://docs.telemetry.mozilla.org/concepts/pipeline/schemas.html),
data engineering has a process of generating schemas for pings and deploying them to BigQuery. The Main Ping table (`telemetry.main`)
is one of those generated tables.

> 
> ## Problems Querying the Main Ping Table: `telemetry.main`

Reviewer: I would personally skip this section divider.

```suggestion
```

> Because we generate a schema with all of the probes for the main ping, the number of columns has grown by a large amount: at this point,
> there are over 10k columns. When querying this table, it causes [BQ to scan a very large number of files](https://console.cloud.google.com/support/cases/detail/25679061?project=moz-fx-data-shared-prod).
> An end-user will see this as a _very_ slow query. Queries remain slow even when filtering on clustered columns, like `sample_id`.

Reviewer 2: ```suggestion
As the number of telemetry probes defined in Firefox grows, so does the number of columns in `telemetry.main`. At this point, we have nearly 10,000 columns, and we ingest many terabytes of main ping data per day. This combination of a very wide schema and a high data volume means that [BQ has to reference metadata for a very large number of files](https://console.cloud.google.com/support/cases/detail/25679061?project=moz-fx-data-shared-prod) each time it runs a query, even if it only ends up needing to read a small fraction of those files. This has led to a problematic experience for iterative analysis use cases.
```

> To reduce the time for querying main ping data, we have included two new tables: `telemetry.main_1pct`, and `telemetry.main_nightly`.

Reviewer 2: ```suggestion
To reduce the time for querying main ping data, we have included two new tables: `telemetry.main_1pct`, and `telemetry.main_nightly`. These can return results for simple queries in a matter of seconds where a logically equivalent query on `telemetry.main` may take minutes.
```

I want to make the benefits more concrete here rather than using language like "very slow query".

> # Main Ping Sample: `telemetry.main_1pct`

Reviewer: ```suggestion
## Main Ping Sample: `telemetry.main_1pct`
```

This table includes a 1% sample across all channels from `telemetry.main`. It includes 6 months of history.

> # Nightly Main Ping Data: `telemetry.main_nightly`

Reviewer: ```suggestion
## Nightly Main Ping Data: `telemetry.main_nightly`
```
